### PR TITLE
Write to the given output path

### LINKED ISSUE #47: Should the JSON CLI flag take a path to output to instead of always using stdout?
Right now we always output to `stdout` and require users to redirect or pipe in their shell. It might be convenient to give an output path and use `--json -` or `--json /dev/stdout` to replicate the current behaviour.

### PR DESCRIPTION
**Pull request by robrix**
This PR:

- [x] Fixes #47:
  - [x] Takes an optional path.
  - [x] Writes JSON to it.

### CODE REVIEW

File: src/graph.rs [hunk 1]
```diff
@@ -87,9 +94,17 @@ impl<'tree> Graph<'tree> {
...
-        print!("{}", s)
+        path.map_or(stdout().write_all(s.as_bytes()), |path| {
+            File::create(path)?.write_all(s.as_bytes())
+        })
+        .with_context(|| {
+            format!(
+                "Failed to write JSON to {}",
+                path.map_or("stdout", |path| path.to_str().unwrap_or("output path"))
+            )
+        })
```
Comment (robrix) [inline src/graph.rs:107]: Can't say I'm in love with this, but trying to abstract over `File` and `Stdout` wasn't going well, and I couldn't figure out appropriate conversions either. Same goes for `&str` and `Cow<str>`, hence some of the faffing about on line 105.

File: src/bin/tree-sitter-graph/main.rs [hunk 2]
```diff
@@ -48,6 +48,13 @@ fn main() -> Result<()> {
         )
         .arg(Arg::with_name("scope").long("scope").takes_value(true))
         .arg(Arg::with_name("json").long("json").takes_value(false))
+        .arg(
+            Arg::with_name("output")
+                .short('o')
+                .long("output")
+                .requires("json")
+                .takes_value(true),
+        )
```
Comment (robrix) [inline src/bin/tree-sitter-graph/main.rs:57]: This was (mercifully) all I needed to allow `-o`/`--output` to be available only when `--json` has been used.

File: src/graph.rs (hunk 1 above)
Comment (patrickt) [reply src/graph.rs:107]: I think this looks really nice, personally.

File: src/bin/tree-sitter-graph/main.rs [hunk 3]
```diff
@@ -116,8 +123,9 @@ fn main() -> Result<()> {
         .with_context(|| format!("Cannot execute TSG file {}", tsg_path.display()))?;
 
     let json = matches.is_present("json");
+    let output_path = matches.value_of("output").map(|str| Path::new(str));
     if json {
-        graph.display_json();
+        graph.display_json(output_path);
```
Comment (patrickt) [inline src/bin/tree-sitter-graph/main.rs:128]: Stick an `unwrap_or(())` here to quiesce the warning.

**Review by patrickt — CHANGES_REQUESTED** [review]

File: src/bin/tree-sitter-graph/main.rs (hunk 3 above)
Comment (robrix) [reply src/bin/tree-sitter-graph/main.rs:128]: Okay, I see it in CI now. I have no idea why this one isn't showing up locally but other warnings are O_o

**Review by patrickt — APPROVED** [review]

**Pull request merged**